arch/arm64/core/reset.S

```diff
@@ -124,6 +124,14 @@ resetwait:
 	ldr	x0, =arm64_cpu_boot_params
 	get_cpu_id x1
 
+	/* RK3568 can not use ldaxr, which will cause a memory fault */
+#ifdef CONFIG_SOC_RK3568
+	ldr		x2, [x0, #BOOT_PARAM_MPID_OFFSET]
+	cmp	x2, #-1
+	bne	1f
+	str		x1, [x0]
+	b primary_core
+#else
```

Comment:
Not a big fan of this for two reasons:
1. I do not want to start polluting the reset code with SOC specific exceptions
2. We are basically replicating the same code with 1 instruction difference

Also what's preventing the compiler to generate others `ldaxr / stlxr` istructions?

Reply:
The reason I modified this part of the code is that the atomic instruction in zephyr 3.4 does not work properly on this SoC.
The bug may be due to the SoC design. Can I directly change it to the 'ldr' or 'str' instruction of the 2.7 release version without adding CONFIG_SOCRK3568? Or using other instructions to replace the `ldaxr / stlxr' ?

Reply:
that was introduced by @povergoing to fix a concurrency issue. But is there any errata about this? If `ldaxr / stlxr` are not working I'm afraid all the atomic ops are also broken?

Reply:
Recently, while working on supporting spintable in Zephyr, I encountered a similar issue. The instruction `stlxr w3, x1, [x0] `executed successfully, and the mpid was written to the memory location. However, w3 remained as 1, resulting in the program jumping to the `1f ` instead of the `primary_core`. @povergoing

Reply:
> that was introduced by @povergoing to fix a concurrency issue. But is there any errata about this? If `ldaxr / stlxr` are not working I'm afraid all the atomic ops are also broken?

Through recent testing, atomic ops works well when mmu and cache are enabled. So other atomic are not broken.